packages/did/src/DidDetails/DidDetails.ts

```diff
@@ -70,33 +66,27 @@ export function getEndpoint(
 /**
  * Generate a signature over the provided input payload, either as a byte array or as a HEX-encoded string.
  *
- * @param did The DID data.
+ * @param did The DID to be used.
  * @param payload The byte array or HEX-encoded payload to sign.
  * @param sign The sign callback to use for the signing operation.
- * @param keyId The key ID to use to generate the signature.
+ * @param keyRelationship (optional) The key relationship, that should be used. Defaults to 'authentication'.
  *
  * @returns The resulting [[DidSignature]].
  */
 export async function signPayload(
-  did: Partial<DidDocument> & Pick<DidDocument, 'authentication' | 'uri'>,
+  did: DidUri,
   payload: Uint8Array | string,
   sign: SignCallback,
-  keyId: DidVerificationKey['id']
+  keyRelationship: VerificationKeyRelationship = 'authentication'
 ): Promise<DidSignature> {
-  const key = getKey(did, keyId)
-  if (!key || !verificationKeyTypes.includes(key.type)) {
-    throw new SDKErrors.DidError(
-      `Failed to find verification key with ID "${keyId}" on DID "${did.uri}"`
-    )
-  }
-  const alg = signatureAlgForKeyType[key.type as VerificationKeyType]
-  const { data: signature } = await sign({
-    publicKey: key.publicKey,
-    alg,
+  const { data: signature, keyUri } = await sign({
```

Comment:
Could we change things in such a way, so that here we simply do `return sign({…`?

Reply:
only, if we expect a hex string from the signing, afaics

Reply:
will be followed up on here: https://github.com/KILTprotocol/ticket/issues/2210